### PR TITLE
Extending Command Manager window

### PR DESCRIPTION
**Pull request by MatBebenek**
Added run command and reset observers.

More details available in Trello ticket: 
https://trello.com/c/6pDWVwYw/16-extending-command-manager-window-code-cleanup

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 1]
```diff
@@ -107,9 +122,30 @@ public void updateCurrentCommandField() {
...
-	public void deleteObservers() {
+	public void deleteObservers(JButton deleteButton) {
+		if (observersDeleted) {
+			resetObservers(deleteButton);
+		} else {
+			this.observerList = this.commandManager.getChangePublisher().getSubscribers();
+			commandManager.getChangePublisher().clearObservers();
+			this.updateObserverListField();
+			observersDeleted = true;
+			deleteButton.setText("Reset observers");
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: the name deleteButton is confusing in this context

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 2]
```diff
@@ -71,19 +74,31 @@ public CommandManagerWindow(DriverCommandManager commandManager) {
...
+		content.add(btnRunCommand, c);
 	}
 
 	private void clearCommand() {
 		commandManager.clearCurrentCommand();
 		updateCurrentCommandField();
 	}
 
+	private void runCommand() {
+		commandManager.runCurrentCommand();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: As a result, of refactor, now CommandManagerWindow is indirectly dependent on DriverManager. Try to fix it. Suggestion: try creating interface(s).

**Review by kowallus — COMMENTED** [review]
ok. Minor design request: reduce coupling.

**Comment by MatBebenek** [comment]
MR was updated according to given comments.

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManager.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import edu.kis.powp.observer.Publisher;
+
+public interface CommandManager {
+
+	void clearCurrentCommand();
+	void runCommand();
+	String getCurrentCommandString();
+	void setCurrentCommand(DriverCommand commandList);
+	Publisher getChangePublisher();
+	void setDriverManager(DriverManager driverManager);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManager.java]: DriverManager dependency is still here. Is this method so important to add it to the interface?

**Review by kowallus — COMMENTED** [review]
The approach is correct, but dependency still exist.

**Comment by MatBebenek** [comment]
You are right. I think we have added it by the mistake.

**Pull request merged**